include/deal.II/base/quadrature_lib.h

```diff
@@ -47,6 +47,58 @@ class QGauss : public Quadrature<dim>
   QGauss(const unsigned int n);
 };
 
+/**
+ * The Gauss-Radau family of quadrature rules for numerical integration.
+ *
+ * This modification of the Gauss quadrature uses one of the two interval end
+ * points as well. Being exact for polynomials of degree <i>2n-2</i>, this
+ * formula is suboptimal by one degrees.
+ *
+ * The quadrature points are the left interval end point plus the roots of the
+ * polynomial <i>(P<sub>n-1</sub>(x)+P<sub>n</sub>(x))/(1+x)</i>
+ * where <i>P<sub>n-1</sub></i> and <i>P<sub>n</sub></i> are Legendre
+ * polynomials. The quadrature weights are <i>2/n<sup>2</sup></i> and
+ * <i>(1-x<sub>i</sub>)/(n<sup>2</sup>(P<sub>n-1</sub>(x<sub>i</sub>))<sup>2</sup>)</i>.
+ *
+ * For the right Gauss-Radau formula the quadrature points are the points
+ * <i>1-x<sub>i</sub></i>, where <i>x<sub>i</sub></i> are the quadrature points
+ * of the left Gauss-Radau formula.
+ *
+ * @note This implementation only features the known exact roots up to
+ * n = 3 quadrature points.
+ *
+ * @sa https://mathworld.wolfram.com/RadauQuadrature.html @sa
+ */
+template <int dim>
+class QGaussRadau : public Quadrature<dim>
+{
+public:
+  /* EndPoint is used to specify which of the two endpoints of the unit interval
+   * is used also as quadrature point
+   */
+  enum EndPoint
+  {
+    /**
+     * Left end point.
+     */
+    left,
+    /**
+     * Right end point.
+     */
+    right
+  };
+  QGaussRadau(const unsigned int n, EndPoint ep = QGaussRadau::left);
+
+  /**
+   * Move constructor. We cannot rely on the move constructor for `Quadrature`,
+   * since it does not know about the additional member `ep` of this class.
+   */
+  QGaussRadau(QGaussRadau<dim> &&) noexcept = default;
+
+private:
+  const EndPoint ep;
```

Comment:
```suggestion
  const EndPoint end_point;
```

Reply:
This is the same notation as in QGaussRadauChebychev,
but I'd be open to change both in this or a second PR.

Reply:
The notation here is the same as in QGaussRadauChebychev,
so I opted for consistency.

Reply:
That would be great! Feel free to leave it to a follow-up patch!

Reply:
As there are various comments concerning the structure and commenting of both QGaussRadau 
and QGaussRadauChebychev I decided to apply all comments to QGaussRadau here.
After merging I will go over the QGaussRadauChebychev class 
and make all applicable changes to unify documentation and interface.